extensions/http-curl/sitetosite/HTTPProtocol.cpp

```diff
@@ -243,37 +242,39 @@ void HttpSiteToSiteClient::closeTransaction(const std::string &transactionID) {
 
   if (it == known_transactions_.end()) {
     return;
-  } else {
-    transaction = it->second;
-    if (transaction->closed_) {
-      return;
-    }
+  }
+
+  transaction = it->second;
+  if (transaction->closed_) {
+    return;
   }
 
   std::string append_str;
-  logger_->log_info("Site to Site closed transaction %s", transaction->getUUIDStr());
+  logger_->log_trace("Site to Site closing transaction %s", transaction->getUUIDStr());
+
+
+  bool data_received = transaction->getDirection() == RECEIVE && (current_code == CONFIRM_TRANSACTION || current_code == TRANSACTION_FINISHED);
 
   int code = UNRECOGNIZED_RESPONSE_CODE;
```

Comment:
There were issues with this. 

In case the RPG was acting as an input port and the remote side (NiFi) provided no flowfiles, the transaction finished without transferring data. This resulted in creating an invalid query here, flooding the logs with errors. 
With this change these transactions are cancelled properly.